service/src/main/java/com/codeforcommunity/processor/PostsProcessor.java

```diff
@@ -0,0 +1,80 @@
+package com.codeforcommunity.processor;
+
+import com.codeforcommunity.api.IPostsProcessor;
+import com.codeforcommunity.database.table.ICommentTable;
+import com.codeforcommunity.database.table.IPostTable;
+import com.codeforcommunity.dto.response.Comment;
+import com.codeforcommunity.dto.response.CommentsResponse;
+import com.codeforcommunity.dto.response.PostsResponse;
+import com.codeforcommunity.dto.response.SinglePostResponse;
+import java.util.List;
+
+/**
+ * The class which will handle all processing of posts. This includes creation, deletion, and
+ * whatever else is needed.
+ */
+public class PostsProcessor implements IPostsProcessor {
+  // Our in-memory 'database table' for holding posts. This will eventually be replaced by the real
+  // db.
+  private final IPostTable postTable;
+  // Our in-memory 'database table' for holding comments.
+  private final ICommentTable commentTable;
+  // A max length we'll have set for our preview.
+  private static final int PREVIEW_MAX_LENGTH = 50;
+
+  public PostsProcessor(IPostTable postTable, ICommentTable commentTable) {
+    // Set the in memory database tables.
+    this.postTable = postTable;
+    this.commentTable = commentTable;
+  }
+
+  /**
+   * Get a substring which is at most {@code previewMaxLength} length but possibly smaller.
+   *
+   * @param body The body to extract the preview from.
+   * @return A string representing the preview.
+   */
+  private static String getBodyPreview(String body) {
```

Comment:
I think it's convention to put private static methods after interface methods... though I'm not sure if that's specified anywhere so I may be wrong

Reply:
It has been made public but I put it at the bottom anyway.